### DISCUSSION TITLE
External Authentication Support

### DISCUSSION THREAD

**Discussion by felix-exon** [osTicket/osTicket#6711] [Ideas]
It would be awesome if you got open-id-connect support for solutions like keycloak.

Benefits:
- SSO
- Central User Management
- Users can use existing logins from multiple applications and manage everything under the same user

**Comment by JediKev** [comment]
@felix-exon 

We have an OAuth2 plugin available that allows you to configure an external auth provider to authenticate users/agents/both via the OAuth2 protocol. If I’m not mistaken open-id is built on top of OAuth2. We have guides available below for Microsoft, Google, and Okta however you can configure any provider as long as they follow the OAuth2 norms:
- https://docs.osticket.com/en/latest/Guides/OAuth2%20Guide.html#authentication

Cheers.
